### DISCUSSION TITLE
Problem unpacking field

### DISCUSSION THREAD

**Discussion by mehrdad2000** [tilln/jmeter-iso8583#100] [General]
Hi
I try to send a message to a mock server, but when it replies, response not unpack and stuck at field 121 got this error:

org.jpos.iso.IFA_LLLLCHAR: Problem unpacking field 121 (java.lang.RuntimeException: Required 3000 but just got 966 bytes) unpacking field=121, consumed=176

Here is my packeger:
```
  <isofield
      id="102"
      length="28"
      name="ACCOUNT IDENTIFICATION 1"
      class="org.jpos.iso.IFA_LLCHAR"/>
  <isofield

  <isofield
      id="121"
      length="5000"
      name="RESERVED PRIVATE USE"
      class="org.jpos.iso.IFA_LLLLCHAR"/>

  <isofield
      id="122"
      length="5000"
      name="RESERVED PRIVATE USE"
      class="org.jpos.iso.IFA_LLLLCHAR"/>

  <isofield
      id="126"
      length="999"
      name="RESERVED PRIVATE USE"
      class="org.jpos.iso.IFA_LLLCHAR"/>
  <isofield


  <isofield
      id="127"
      length="5000"
      name="RESERVED PRIVATE USE"
      class="org.jpos.iso.IFA_LLLLCHAR"/>
  <isofield
```

Here is the log [error-unpack.txt](https://github.com/user-attachments/files/16161893/error-unpack.txt)

Any idea?
Thanks

**Comment by tilln** [comment]
Most likely your packager config is wrong.
I can't help without the entire config file though.

With standard ISO8583 '87 field lengths, I can see how the error would happen:

```
MTI 0610
Bitmap E03801800A8100040000000004000086

11100000001110000000000110000000000010101000000100000000000001000000000000000000000000000000000000000100000000000000000010000110
1,2,3     11,12,13     24,25        37,39,41   48            62                                      102                121  126,127

DE2  (16)5859836490000732
DE3  890000
DE11 432013
DE12 154010
DE13 0224
DE24 100
DE25 03
DE37 839203026130
DE39 00
DE41 00091810
DE48 (063)F3FF40011111000O0640N13AutoGenerated4040414030420N029999N044011
DE62 000
DE102 (02)80
DE121 (3000)??? 00009251585983     00090026C6600101040501140304209900172FF8000008A7A0000111111111110465638406     0103...04....07431029000000000912222222200007500000001000000000000000000000000000036E00000000000000000000000000000000000  000000000734303030303030323030303033353330322E32015B00080030300001023031014F110230310D00303030303030303030303038002A01300026250F00EF001E0000000000129002119004119005119006119009119160110000000000000000000000F89001103538353938333634393030303037333290020847EC728EF88E8F3490030D4175746F47656E657261746564900408A7A332608D51343F900508133F69189277D9DA900610A54F549F9AB9F498F817912A0DB53F67900803313031900908CBE8258EA8417E559011023031902003CAD3CA903004CAD3CAED90400A3034363536333834303690500A313131313131313131319120033130309130063433323031339140043131313191500635383539383391602208D81DD8932D6E78E907F57E5DAFA9E15DB06D274C78FFFD47EC728EF88E8F343031917001319200083134303330343230921004303634309230073530303030303014DF3E4C1173E35390ACD95792D956CBF419FEF000
```

**Reply by mehrdad2000** [reply]
@tilln I’ll post packager tomorrow here.
But for now:
1- how the error would happen? (3 question mark in beginning of DE121?)
2- do you use specific script or tools to parse messages like post in previous reply? If so would you please share it?
3- as i see DE62 must be 0000, and DE102 length must be (28). Seems shift 1 character!

FYI: it would be great if it’s add as feature for debugging purpose, e.g normally when jmeter-iso8583 send message, if it has issue in respond not show parse message in “view result tree response tab”. So if it parse like you mentioned in previous reply in log of jmeter in debug mode, troubleshooting would be easier. Even as post jrs223 groovy script would be good.

Thanks

**Reply by tilln** [reply]
> 1- how the error would happen? (3 question mark in beginning of DE121?)

Obviously, the 3000 is wrong as the field length, due to some of the previous fields using incorrect lengths and some digits begin shifted.

> 2- do you use specific script or tools to parse messages like post in previous reply? If so would you please share it?

No tools, I just did this by hand.

> 3- as i see DE62 must be 0000, and DE102 length must be (28). Seems shift 1 character!

Ok, that helps, but there are still more issues, as now DE126 has invalid length:
```
DE62 0000
DE102 (28)0300000009251585983     0009
DE121 (0026)C6600101040501140304209900
DE126 (172F)??? F8000008A7A0000111111111110465638406     0103...04....07431029000000000912222222200007500000001000000000000000000000000000036E00000000000000000000000000000000000  000000000734303030303030323030303033353330322E32015B00080030300001023031014F110230310D00303030303030303030303038002A01300026250F00EF001E0000000000129002119004119005119006119009119160110000000000000000000000F89001103538353938333634393030303037333290020847EC728EF88E8F3490030D4175746F47656E657261746564900408A7A332608D51343F900508133F69189277D9DA900610A54F549F9AB9F498F817912A0DB53F67900803313031900908CBE8258EA8417E559011023031902003CAD3CA903004CAD3CAED90400A3034363536333834303690500A313131313131313131319120033130309130063433323031339140043131313191500635383539383391602208D81DD8932D6E78E907F57E5DAFA9E15DB06D274C78FFFD47EC728EF88E8F343031917001319200083134303330343230921004303634309230073530303030303014DF3E4C1173E35390ACD95792D956CBF419FEF000
```

> FYI: it would be great if it’s add as feature for debugging purpose

You get some packager logging output when starting JMeter with these command line arguments
`-L nz.co.breakpoint.jmeter.iso8583=DEBUG -J jmeter.iso8583.q2PackagerLogging=true`.

**Reply by mehrdad2000** [reply]
Here is the packager 
[iso87ascii.xml.txt](https://github.com/user-attachments/files/16201228/iso87ascii.xml.txt)

**Reply by tilln** [reply]
Not sure how else I can help you here.

Your DE62 is defined as `IFA_LLLCHAR` which takes us back to the above error:

```
DE62 (000)
DE102 (02)80
DE121 (3000)<---
```

**Reply by mehrdad2000** [reply]
@tilln You are right it's my fault, didn't pay attention to that DE! :) 
set DE 62 IF_CHAR 4, now it works like charm!

Thanks so much for your endless support.
